src/toplevel.c

```diff
@@ -865,6 +863,7 @@ JL_DLLEXPORT jl_value_t *jl_toplevel_eval_flex(jl_module_t *JL_NONNULL m, jl_val
 
     if (head == jl_module_sym) {
         jl_value_t *val = jl_eval_module_expr(m, ex);
+        ct->world_age = jl_atomic_load_acquire(&jl_world_counter);
```

Comment:
Seems really awkward (and error prone) to now have this function failing to restore world_age on return, since normally we expect this value to be scoped to the local function, and some callers of this function (e.g. the REPL implementation) would become invalid if that changes
```suggestion
```